knowledge-base/grid-rowcollection-nullreference.md

```diff
@@ -0,0 +1,53 @@
+---
+title: Null Reference Expection from the GridRowCollection
+description: Fixing a Null Reference Exception Thrown from the GridRowCollection.
+type: troubleshooting
+page_title: Null Reference Expection from the GridRowCollection
+slug: grid-kb-nullref-gridrowcollection
+position: 
+tags: 
+ticketid: 1643250, 1642910, 1604815, 1574971, 1577863, 1580684
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product</td>
+			<td>Grid for Blazor</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+
+I upgraded the version of the Telerik UI from Blazor components from `<3.0.0` to `>3.0.0` and I get a `NullReferenceException` from the `GridRowCollection` class.
+
+## Error Message
+
+>warning Error: System.NullReferenceException: Object reference not set to an instance of an object.
+   at Telerik.Blazor.Components.Grid.GridRowCollection`1.BuildRenderTree(RenderTreeBuilder __builder)
+
+## Cause
+
+The Telerik UI Grid for Blazor throws a `NullReferenceException` from the `GridRowCollection` class when both the `Data` parameter and `OnRead` event handler are defined:
+
+````CSHTML
+
+<TelerikGrid Data="@DataCollection" OnRead="@OnReadHandler">
+    ...
+</TelerikGrid>
+````
+
+## Solution
+
+Remove either the `Data` parameter or the `OnRead` event handler.
```

Comment:
If the developer has used `OnRead` for some reason, it's unlikely they will go back to `Data`. Thus the primary recommendation can be phrased in a different way. For example, focus on the new way to use `OnRead` and link the relevant documentation here. The "See Also" links are less visible by definition.